cmd/vic-machine/configure/configure.go

```diff
@@ -0,0 +1,245 @@
+	var flags []cli.Flag
+	for _, f := range [][]cli.Flag{target, id, compute, iso, util, debug} {
+		flags = append(flags, f...)
+	}
+
+	return flags
+}
+
+func (c *Configure) processParams() error {
+	defer trace.End(trace.Begin(""))
+
+	if err := c.HasCredentials(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Configure) Run(clic *cli.Context) (err error) {
+	// urfave/cli will print out exit in error handling, so no more information in main method can be printed out.
+	defer func() {
+		err = common.LogErrorIfAny(clic, err)
+	}()
+
+	// process input parameters, this should reuse same code with create command, to make sure same options are provided
+	if err = c.processParams(); err != nil {
+		return err
+	}
+
+	if c.Debug.Debug > 0 {
+		log.SetLevel(log.DebugLevel)
+		trace.Logger.Level = log.DebugLevel
+	}
+
+	if len(clic.Args()) > 0 {
+		log.Errorf("Unknown argument: %s", clic.Args()[0])
+		return errors.New("invalid CLI arguments")
+	}
+
+	// TODO: add additional parameter processing, reuse same code with create command as well
+
+	if c.upgrade {
+		// verify upgrade required parameters here
+	}
+
+	log.Infof("### Configuring VCH ####")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	validator, err := validate.NewValidator(ctx, c.Data)
+	if err != nil {
+		log.Errorf("Configuring cannot continue - failed to create validator: %s", err)
+		return errors.New("configure failed")
+	}
+	_, err = validator.ValidateTarget(ctx, c.Data)
+	if err != nil {
+		log.Errorf("Configuring cannot continue - target validation failed: %s", err)
+		return errors.New("configure failed")
+	}
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, c.Force)
+
+	var vch *vm.VirtualMachine
+	if c.Data.ID != "" {
+		vch, err = executor.NewVCHFromID(c.Data.ID)
+	} else {
+		vch, err = executor.NewVCHFromComputePath(c.Data.ComputeResourcePath, c.Data.DisplayName, validator)
+	}
+	if err != nil {
+		log.Errorf("Failed to get Virtual Container Host %s", c.DisplayName)
+		log.Error(err)
+		return errors.New("configure failed")
+	}
+
+	log.Infof("")
+	log.Infof("VCH ID: %s", vch.Reference().String())
+
+	if c.ResetInProgressFlag {
+		if err = vch.SetVCHUpdateStatus(ctx, false); err != nil {
+			log.Error("Failed to reset UpdateInProgress flag")
+			log.Error(err)
+			return errors.New("configure failed")
+		}
+		log.Infof("Reset UpdateInProgress flag successfully")
+		return nil
+	}
+
+	var vchConfig *config.VirtualContainerHostConfigSpec
+	if c.upgrade {
+		vchConfig, err = executor.FetchAndMigrateVCHConfig(vch)
+	} else {
+		vchConfig, err = executor.GetVCHConfig(vch)
+	}
+	if err != nil {
+		log.Error("Failed to get Virtual Container Host configuration")
+		log.Error(err)
+		return errors.New("configure failed")
+	}
+
+	// TODO: Convert guestinfo *VirtualContainerHost back to *Data, decrypt secret data
+	oldData := &data.Data{}
+	// oldData, err := converter.GtoD(vchConfig)
```

Comment:
This is the comment intentionally left here. Those code is not there yet, but I want to have them added here.